stargazer/stargazer.py

```diff
@@ -216,16 +216,78 @@ def append_notes(self, append):
         assert type(append) == bool, 'Please input True/False'
         self.notes_append = append
 
-    # Begin HTML render functions
-    def render_html(self):
+    def render_html(self, *args, **kwargs):
+        return HTMLRenderer(self).render(*args, **kwargs)
+
+    def _repr_html_(self):
+        return self.render_html()
+
+    def render_latex(self, *args, **kwargs):
+        return LaTeXRenderer(self).render(*args, **kwargs)
+
+
+class Renderer:
+    """
+    Base class for renderers to specific formats. Only meant to be subclassed.
+    """
+    def __init__(self, table):
+        """
+        Initialize a new renderer.
+        
+        "table": Stargazer object to render
+        """
+
+        self.table = table
+
+    def __getattribute__(self, key):
+        """
+        Temporary fix while we better organize how a Stargazer table stores
+        parameters: just retrieve them transparently as attributes of the
+        Stargazer table object.
+        """
+
+        try:
+            return object.__getattribute__(self, key)
+        except AttributeError as exc:
+            if hasattr(self.table, key):
+                return getattr(self.table, key)
+            else:
+                raise exc
+
+    def get_sig_icon(self, p_value, sig_char='*'):
+        if p_value is None:
+            return ''
+        if p_value >= self.sig_levels[0]:
+            return ''
+        elif p_value >= self.sig_levels[1]:
+            return sig_char
+        elif p_value >= self.sig_levels[2]:
+            return sig_char * 2
+        else:
+            return sig_char * 3
+
+    def _float_format(self, value):
+        """
+        Format value to string, using the precision set by the user.
+        """
+        if value is None:
+            return ''
+
+        return '{{:.{prec}f}}'.format(prec=self.sig_digits).format(value)
+
+
+class HTMLRenderer(Renderer):
+    fmt = 'html'
+
+    def render(self):
         html = ''
```

Comment:
Minor point, but seems like this and a few other places could start with `html = self.generate_header()` rather than an empty string. Or is this a best practice I'm unaware of?

Great PR!

Reply:
I had probably copied some style... but not one that I would recommend: changed as suggested.